src/components/FileUpload/index.tsx

```diff
@@ -28,56 +30,65 @@ const FileUpload = (props: TFileUploadProps): JSX.Element | null => {
     }
   }
 
-  const getFileType = () => {
-    if (file && file.type) {
-      return file.type.split('/')[1]
-    }
-    return ''
-  }
-
   const updateInForm = useCallback(
-    (value: File | null) => {
+    (values: File[] | null) => {
       if (name && setFieldValue) {
-        setFieldValue(name, value, { shouldValidate: !!value })
+        setFieldValue(name, values as TFormValue, { shouldValidate: !!values })
       }
     },
     [name, setFieldValue]
   )
 
-  const getFormattedValue = (file: File) => {
-    const reader = new FileReader()
-    reader.onloadend = () => {
-      console.log(reader.result)
-    }
-    reader.readAsDataURL(file)
-    return reader
-  }
+  const createAcceptFormat = useMemo(
+    () => (allowedTypes.includes('*') ? '*' : allowedTypes.map((type) => '.' + type).join(',')),
+    [allowedTypes]
+  )
 
-  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target?.files && event?.target?.files[0]) {
-      setFile(event?.target?.files[0])
-      updateInForm(event?.target?.files[0])
+  const getFormattedValues = (files: File[]) => {
+    const readers: FileReader[] = []
 
-      if (getFiles) {
-        if (toBase64) {
-          getFormattedValue(event?.target?.files[0])
-        } else {
-          getFiles(event?.target?.files[0])
-        }
+    for (let i = 0; i < files.length; i++) {
+      const reader = new FileReader()
+      reader.onloadend = () => {
+        console.log(reader.result)
       }
+      reader.readAsDataURL(files[i])
+      readers.push(reader)
     }
+
+    return readers
   }
 
-  const handleFileRemove = useCallback(() => {
-    setFile(null)
-    updateInForm(null)
-  }, [updateInForm])
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const selectedFiles = event.target?.files
+      if (selectedFiles) {
+        const fileArray = Array.from(selectedFiles)
+        setFiles([...files, ...fileArray])
+        updateInForm(fileArray)
+        if (getFiles) {
+          if (toBase64) {
+            getFormattedValues(fileArray)
```

Comment:
what is the functionality of this row? maybe it should assign return value to fileArray?